protocols/perf/src/bin/perf-server.rs

```diff
@@ -0,0 +1,131 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use clap::Parser;
+use futures::{executor::block_on, future::Either, StreamExt};
+use libp2p_core::{
+    identity, muxing::StreamMuxerBox, transport::OrTransport, upgrade, PeerId, Transport,
+};
+use libp2p_dns::DnsConfig;
+use libp2p_swarm::{SwarmBuilder, SwarmEvent};
+use log::{error, info};
+
+#[derive(Debug, Parser)]
+#[clap(name = "libp2p perf server")]
+struct Opts {}
+
+fn main() {
+    env_logger::init();
+
+    let _opts = Opts::parse();
+
+    // Create a random PeerId
+    let local_key = identity::Keypair::generate_ed25519();
+    let local_peer_id = PeerId::from(local_key.public());
+    println!("Local peer id: {local_peer_id:?}");
+
+    let transport = {
+        let tcp =
+            libp2p_tcp::async_io::Transport::new(libp2p_tcp::Config::default().port_reuse(true))
+                .upgrade(upgrade::Version::V1Lazy)
+                .authenticate(
+                    libp2p_noise::NoiseAuthenticated::xx(&local_key)
+                        .expect("Signing libp2p-noise static DH keypair failed."),
+                )
+                .multiplex(libp2p_yamux::YamuxConfig::default());
+
+        let quic = {
+            let mut config = libp2p_quic::Config::new(&local_key);
+            config.support_draft_29 = true;
+            libp2p_quic::async_std::Transport::new(config)
+        };
+
+        let dns = block_on(DnsConfig::system(OrTransport::new(quic, tcp))).unwrap();
+
+        dns.map(|either_output, _| match either_output {
+            Either::Left((peer_id, muxer)) => (peer_id, StreamMuxerBox::new(muxer)),
+            Either::Right((peer_id, muxer)) => (peer_id, StreamMuxerBox::new(muxer)),
+        })
+        .boxed()
+    };
+
+    let mut swarm = SwarmBuilder::with_async_std_executor(
+        transport,
+        libp2p_perf::server::behaviour::Behaviour::default(),
+        local_peer_id,
+    )
+    .substream_upgrade_protocol_override(upgrade::Version::V1Lazy)
+    .build();
+
+    swarm
+        .listen_on("/ip4/0.0.0.0/tcp/4001".parse().unwrap())
+        .unwrap();
+
+    swarm
+        .listen_on("/ip4/0.0.0.0/udp/4001/quic-v1".parse().unwrap())
+        .unwrap();
+
+    block_on(async {
+        loop {
+            match swarm.next().await.unwrap() {
+                SwarmEvent::NewListenAddr { address, .. } => {
+                    info!("Listening on {:?}", address);
```

Comment:
```suggestion
                    info!("Listening on {}", address);
```
 Hard to copy-paste otherwise.

Reply:
ce576555